hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -489,6 +473,26 @@ private boolean isLegalToAdjust(long balance, long adjustment) {
 		return (balance + adjustment >= 0);
 	}
 
+	private ResponseCodeEnum plausibilityOf(BalanceChange change) {
+		final var id = change.accountId();
+		if (!exists(id) || isSmartContract(id)) {
+			return INVALID_ACCOUNT_ID;
+		}
+		if ((boolean) accountsLedger.get(id, IS_DELETED)) {
+			return ACCOUNT_DELETED;
+		}
+		if (isDetached(id)) {
+			return ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
+		}
```

Comment:
This code section is a good candidate for a function to check an AccountID's validity.

Reply:
Very interesting question...will give this some thought, and probably add a `TransactionalLedger.validate()` method it in a future PR.